Documentation/sources/content-card-ui/tutorial/displaying-content-cards.md

```diff
@@ -0,0 +1,139 @@
+# Fetch and Display Content Cards
+
+This tutorial explains how to fetch and display content cards in your application.
+
+## Pre-requisites
+
+[Integrate and register AEPMessaging extension](https://developer.adobe.com/client-sdks/edge/adobe-journey-optimizer/#implement-extension-in-mobile-app) in your app.
+
+## Fetch Content Cards
+
+To fetch the content cards for the surfaces configured in [Adobe Journey Optimizer](https://business.adobe.com/products/journey-optimizer/adobe-journey-optimizer.html) campaigns, call the [updatePropositionsForSurfaces](https://developer.adobe.com/client-sdks/edge/adobe-journey-optimizer/code-based/api-reference/#updatepropositionsforsurfaces) API. It's recommended to batch requests for multiple surfaces in a single API call when possible. The returned content cards are cached in-memory by the Messaging extension and persist through the application's lifecycle.
+
+```kotlin
+val surfaces = mutableListOf<Surface>()
+val surface = Surface("homepage")
+Messaging.updatePropositionsForSurfaces(surfaces)
+```
+
+## Retrieve Content Cards
+
+To retrieve the content cards for a specific surface, call `getContentCardsUI`. This API returns a [flow](https://developer.android.com/kotlin/flow) of [AepUI](../public-classes/aepui.md) objects representing content cards for which the user is qualified.
+
+`AepUI` objects are created only for content cards with templates recognized by the Messaging extension. The flow of `AepUI` objects may contain multiple content card template types.
+
+```kotlin
+// create a view model or reuse existing one to hold the aepUIList
+class AepContentCardViewModel(private val contentCardUIProvider: ContentCardUIProvider) : ViewModel() {
```

Comment:
We can provide a few more options on how to retrieve the flow of AepUI if we want. The wiki has some examples
https://wiki.corp.adobe.com/display/adms/Using+Content+Cards+with+UI+on+Android#UsingContentCardswithUIonAndroid-RetrieveContentCards